src/components/_tests_/EnumSwitch.spec.ts

```diff
@@ -0,0 +1,107 @@
+        alertMessage: "",
+        enumList: [],
+        enumAttributeValue: "Chicken",
+        onClickAction: jasmine.createSpy("onClick"),
+        status: "enabled",
+        bootstrapStyle: "default"
+    };
+
+    it("renders the structure correctly", () => {
+        const enumswitch = shallowRenderSwitch(defaultProps);
+
+        expect(enumswitch).toBeElement(
+            createElement("div", { className: "form-validation" },
+                createElement("div", { className: "widget-enum-switch form-control" },
+                    createElement(EnumButton, {
+                        status: defaultProps.status,
+                        bootstrapStyle: defaultProps.bootstrapStyle,
+                        position: enumswitch.state().position,
+                        visibility: enumswitch.state().visibility,
+                        width: enumswitch.state().width,
+                        height: enumswitch.state().height,
+                        getButtonNode: jasmine.any(Function)
+                    })
+                ),
+                createElement(Alert, { message: defaultProps.alertMessage || "", bootstrapStyle: "danger" })
+            ));
+    });
+
+    it("should not create button if there is no default atrribute", () => {
+        defaultProps.enumAttributeValue = "";
+        const enumswitch = shallowRenderSwitch(defaultProps);
+
+        expect(enumswitch).toBeElement(
+            createElement("div", { className: "form-validation" },
+                createElement("div", { className: "widget-enum-switch form-control" }),
+                createElement(Alert, { message: defaultProps.alertMessage || "", bootstrapStyle: "danger" })
+            ));
+    });
+
+    it("should slide enum toggle onClick", () => {
+        const enumswitch = fullRenderSwitch(defaultProps);
+        const enumSwitchInstance = enumswitch.instance() as any;
+        const enumSlider = spyOn(enumSwitchInstance, "enumToggleSlider").and.callThrough();
+        defaultProps.enumAttributeValue = "Salami";
+        enumSwitchInstance.componentDidUpdate(defaultProps);
+
+        expect(enumSlider).toHaveBeenCalled();
+    });
+
+    it("should remove events when unmounting", () => {
+        const enumswitch = shallowRenderSwitch(defaultProps);
+        const enumSwitchInstance = enumswitch.instance() as any;
+        const enumSlider = spyOn(enumSwitchInstance, "removeEvents").and.callThrough();
+        enumSwitchInstance.componentWillUnmount();
+        enumswitch.unmount();
+
+        expect(enumSlider).toHaveBeenCalled();
+    });
+
+    describe("that is enabled", () => {
+        it("should have the enabled class", () => {
+            const enumswitch = shallowRenderSwitch(defaultProps);
+
+            expect(enumswitch.hasClass("disabled")).not.toBe(true);
+        });
+    });
+
+    describe("that is disabled", () => {
+        it("should not have the enabled class", () => {
+            defaultProps.status = "disabled";
```

Comment:
If you are sharing variables in unit tests. it should not change. If it changes create a copy and use in your tests